jaxrs-clients/src/main/java/com/palantir/remoting/jaxrs/FeignJaxRsClientBuilder.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting.jaxrs;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.kristofa.brave.ClientRequestInterceptor;
+import com.github.kristofa.brave.ClientResponseInterceptor;
+import com.github.kristofa.brave.ClientTracer;
+import com.github.kristofa.brave.Sampler;
+import com.github.kristofa.brave.ThreadLocalServerClientAndLocalSpanState;
+import com.github.kristofa.brave.http.DefaultSpanNameProvider;
+import com.github.kristofa.brave.okhttp.BraveOkHttpRequestResponseInterceptor;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.net.InetAddresses;
+import com.palantir.ext.brave.SlfLoggingSpanCollector;
+import com.palantir.remoting.http.BackoffStrategy;
+import com.palantir.remoting.http.FailoverFeignTarget;
+import com.palantir.remoting.http.GuavaOptionalAwareContract;
+import com.palantir.remoting.http.NeverRetryingBackoffStrategy;
+import com.palantir.remoting.http.ObjectMappers;
+import com.palantir.remoting.http.SlashEncodingContract;
+import com.palantir.remoting.http.UserAgentInterceptor;
+import com.palantir.remoting.http.errors.FeignSerializableErrorErrorDecoder;
+import feign.Contract;
+import feign.Feign;
+import feign.InputStreamDelegateDecoder;
+import feign.InputStreamDelegateEncoder;
+import feign.Logger;
+import feign.OptionalAwareDecoder;
+import feign.Request;
+import feign.TextDelegateDecoder;
+import feign.TextDelegateEncoder;
+import feign.codec.Decoder;
+import feign.codec.Encoder;
+import feign.jackson.JacksonDecoder;
+import feign.jackson.JacksonEncoder;
+import feign.jaxrs.JaxRsWithHeaderAndQueryMapContract;
+import feign.okhttp3.OkHttpClient;
+import feign.slf4j.Slf4jLogger;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.List;
+import java.util.Random;
+import org.joda.time.Duration;
+
+final class FeignJaxRsClientBuilder extends ClientBuilder {
```

Comment:
yeah, the setup is a little strange at the moment: the more abstract thing (jaxrs-clients) depends on the the less abstract thing (feign-config). I'm not sure how much abstraction we actually need here: do we ever want to support multiple different types of jaxrs clients and make swappable?